File: javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  // 0, 1, 1, 2, 3, 5, 8
  let fibArr = [0, 1]

  let max = 0
  for(let i=2; i <= num; i++){
    fibArr[i] = fibArr[i-2] += fibArr[i-1]
    max<fibArr[i]? max = fibArr[i]:null
  }
  return max
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  # type your code in here
  fibArr = [0, 1]
  return fibArr[num] if num < 2

  (num-1).times do 
    sum = fibArr[0] + fibArr[1]
    fibArr = [fibArr[1], sum]
  end

  fibArr.last  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
